File: src/main/kotlin/com/xs0/gqlktx/types/gql/GValueType.java
package com.xs0.gqlktx.types.gql;

public abstract class GValueType extends GBaseType {
    protected GValueType(String name) {
        super(name);
    }

    @Override
    public boolean validAsArgumentType() {
        return true;
    }

    @Override
    public boolean validAsQueryFieldType() {
        return true;
    }
}

File: src/main/kotlin/com/xs0/gqlktx/types/gql/GValueType.kt
package com.xs0.gqlktx.types.gql

abstract class GValueType protected constructor(name: String) : GBaseType(name) {

    override val validAsArgumentType: Boolean
        get() {
            return true
        }

    override val validAsQueryFieldType: Boolean
        get() {
            return true
        }
}
